test/Bond.spec.ts

```diff
@@ -284,6 +284,84 @@ describe("Bond", () => {
           });
         });
       });
+      describe("#withdrawExcessPayment", async () => {
+        // excess payment should be withdrawable if bond is overpaid
+        // excess payment should be withdrawable when bonds are burned
+        // excess payment should be withdrawable when bonds are converted
+        // can't withdraw if payment is not excesss
+        // can not withdraw if payment equals payment amount
+        // user can create bond, pay 50%, burn bonds, then withdraw payment
+        //
+        describe("non-convertible", async () => {
+          beforeEach(async () => {
+            bond = bondWithTokens.nonConvertible.bond;
+            config = bondWithTokens.nonConvertible.config;
+            await paymentToken.approve(
+              bond.address,
+              ethers.constants.MaxUint256
+            );
+          });
+
+          it("can't withdraw if payment is not excesss", async () => {
+            expect(await bond.amountOverPaid()).to.equal(0);
+            await paymentToken.transfer(bond.address, await bond.amountOwed());
+            expect(await bond.amountOverPaid()).to.equal(0);
+          });
+
+          it("excess payment should be withdrawable if bond is overpaid", async () => {
+            await paymentToken.transfer(
+              bond.address,
+              (await bond.amountOwed()).add(1)
+            );
+            expect(await bond.amountOverPaid()).to.equal(1);
+            await bond.withdrawExcessPayment();
+            expect(await bond.amountOverPaid()).to.equal(0);
+          });
+          it("excess payment should be withdrawable when bonds are redeemed", async () => {
+            const bonds = await bond.balanceOf(owner.address);
+            const fullpayment = await bond.amountOwed();
+            await paymentToken.transfer(bond.address, fullpayment.mul(2));
+            expect(await bond.amountOverPaid()).to.equal(fullpayment);
+            const [paymentOnRedeem] = await bond.previewRedeemAtMaturity(bonds);
+            expect(paymentOnRedeem).to.equal(fullpayment);
+            await bond.redeem(bonds);
+
+            expect(await bond.amountOverPaid()).to.equal(fullpayment);
+            await bond.withdrawExcessPayment();
+            expect(await bond.amountOverPaid()).to.equal(0);
+          });
+          it("user can create bond, pay 50%, burn bonds, then withdraw payment", async () => {
+            const bonds = await bond.balanceOf(owner.address);
```

Comment:
this is sort of weird because it's not using `totalSupply` and in the line below getting the total amount owed.

Reply:
I would do `bond.totalSupply()`, but i agree it's not any different in this test.